src/plugins/dashboard/public/application/utils/get_no_items_message.tsx

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { Fragment } from 'react';
+import { FormattedMessage } from '@osd/i18n/react';
+import { EuiButton, EuiEmptyPrompt, EuiLink } from '@elastic/eui';
+import { ApplicationStart } from 'opensearch-dashboards/public';
+
+export const getNoItemsMessage = (
+  hideWriteControls: boolean,
+  createItem: () => void,
+  application: ApplicationStart
+) => {
+  if (hideWriteControls) {
+    return (
+      <div>
+        <EuiEmptyPrompt
+          iconType="visualizeApp"
+          title={
+            <h1 id="dashboardListingHeading">
```

Comment:
This id is a hint that this is probably specific to the dashboard_listing component and not necessarily a general purpose dashboard utility. Also, why do we need the id?

Reply:
Yea this function will only be used be the dashboard_listing component. 

Even though it is not going to be used elsewhere, i think it is more organized to include these helper functions in the utils folder? The `Visualize` plugin also includes this function in a separate file under utils folder; and include the components file such as visualize_editor, visualize_listing, visualize_top_nav in their own components folder; and i plan to follow the same file structure.

Reply:
That's fine - I like consistency with the current code base, even if I disagree with it (which I do, in this case 😉).